### PR TITLE
V2 remove center on primitives

### PR DESCRIPTION
**Pull request by garrlker**
### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests? -- Mostly, 3 tests fail

> Note: please do NOT include build files (those generate by the build-xxx commands) with your PR,

Thank you for your help in advance, much appreciated !

### CODE REVIEW

File: packages/modeling/src/primitives/roundedCuboid.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ const {vec2, vec3} = require('../math')
 
 const {geom3, poly3} = require('../geometry')
 
-const createCorners = (center, size, radius, segments, slice, positive) => {
```
Comment (z3dev) [inline packages/modeling/src/primitives/roundedCuboid.js]: You need to change this back, as createCorners() is an internal function only. The center parameter is key to the generation of the polygons.

File: packages/modeling/src/primitives/star.js [hunk 2]
```diff
@@ -12,7 +12,8 @@ const getRadiusRatio = (vertices, density) => {
   return 0
 }
 
-const getPoints = (vertices, radius, startAngle, center) => {
+const getPoints = (vertices, radius, startAngle) => {
```
Comment (z3dev) [inline packages/modeling/src/primitives/star.js]: Please revert this change as getPoints() is an internal routine. The center is used when creating the 2D points.

File: packages/modeling/src/primitives/roundedCuboid.js (hunk 1 above)
Comment (garrlker) [reply packages/modeling/src/primitives/roundedCuboid.js]: Done, this did fix the broken test as well.

File: packages/modeling/src/primitives/star.js (hunk 2 above)
Comment (garrlker) [reply packages/modeling/src/primitives/star.js]: Done, this did fix the broken test as well.

**Comment by z3dev** [comment]
Good. Good. Those changes helped.

The tests are only failing for expand. I’ll take a detailed long at the code tomorrow.
(FYI, I live in Japan)

**Comment by z3dev** [comment]
@garrlker please adjust the code in operations/expansions/expandShell.js, line 200
this function uses sphere() to build as set of corners for the 3D expansion, and uses the center option. a change is required.

```
    let corner = sphere({
      center: [vertex[0], vertex[1], vertex[2]],
      radius: delta,
      segments: segments,
      axes: [xaxis, yaxis, zaxis]
    })
```

**Comment by z3dev** [comment]
@garrlker great stuff. I think we are getting close.

I suspect that the dxf-deserializer is using center. This maybe a little more difficult.

**Comment by garrlker** [comment]
I did find where dxf was using a center param and I changed it to use the function but it's still failing.

I'll look at it some more tomorrow.

**Comment by z3dev** [comment]
See...

packages/io/dxf-deserializer/instantiate.js
packages/io/dxf-deserializer/translate.js

**Comment by z3dev** [comment]
Also SVG deserializer... See...

packages/io/svg-deserializer/shapesMapJscad.js
packages/io/svg-deserializer/shapesMapGeometry.js

**Comment by kaosat-dev** [comment]
@garrlker thanks for the awesome hard work ! Getting there :)
@z3dev & @garrlker , I feel stupid for pointing this out now so late, but our center() function's api is horrible : ```center({center: value}, operands)``` should be ```center(value, operands)``` (or even better ```centerAt(value, operands)```
Since that is another seperate change It is perhaps best to leave it for another PR though ?

**Comment by z3dev** [comment]
> center() function's api is horrible : `center({center: value}, operands)` should be `center(value, operands)` (or even better `centerAt(value, operands)`
> Since that is another seperate change It is perhaps best to leave it for another PR though ?

Another PR, I think.

**Comment by z3dev** [comment]
@garrlker still there?

DXF-deserializer and svg-deserializer need changes. Can you finish this week?

**Comment by garrlker** [comment]
@kaosat-dev 
No problem!
I think another PR would be best :smile: 

@z3dev 
Sorry, I had a busy weekend and wasn't able to hop on.
I've pushed those changes, but the `path2toDxf.test.js` LWPOLYLINE test is still failing.

When I skip that test all others pass, so I think there is a good shot of finishing this up by the end of this week!

**Comment by z3dev** [comment]
> I've pushed those changes, but the `path2toDxf.test.js` LWPOLYLINE test is still failing.

Ah... polylines... ok. There’s probably a small change required. I’ll try to locate the lines needing change today..

**Comment by z3dev** [comment]
Some changes are required to io/dxf-serializer/tests/path2ToDxf.test.js 

This test uses center()

**Comment by garrlker** [comment]
I think I already converted that file's tests to use the correct `center` API.

**Comment by z3dev** [comment]
OK. The problem is that arc() creates an arc which has a center at the [x,y] provided. This is NOT the same as centering the shape at [x,y].  So, these two statements are NOT the same.

```
  let path2 = primitives.arc({ center: [5, 5], endAngle: 45 })
  let path2 = transforms.center({ center: [5,5,0] }, primitives.arc({ endAngle: 45 }));
```

@kaosat-dev any thoughts about this use of the center option?

**Comment by garrlker** [comment]
That makes sense.

I'll roll back the `arc` changes.

**Comment by kaosat-dev** [comment]
@z3dev & @garrlker : agreed & my 2 cents is, for arc, the meaning of center is not quite the same so it is ok to keep the center option there, and roll that part back (sorry @garrlker !)

**Comment by garrlker** [comment]
It's no problem!
I rolled back the arc changes and all tests are passing.

**Comment by z3dev** [comment]
> I rolled back the arc changes and all tests are passing.

Awesome. And thanks!
We will do a quick review.

**Comment by z3dev** [comment]
@kaosat-dev please review.
